pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -258,10 +272,16 @@ func (dsw *desiredStateOfWorld) MarkVolumesReportedInUse(
 }
 
 func (dsw *desiredStateOfWorld) DeletePodFromVolume(
-	podName types.UniquePodName, volumeName v1.UniqueVolumeName) {
+	podName types.UniquePodName, volumeName v1.UniqueVolumeName, volumeSpec *volume.Spec) {
```

Comment:
Is this `volumeSpec` parameter necessary here? Maybe we can get it from the `podToMount` struct, see the code [here](https://github.com/kubernetes/kubernetes/blob/7253c35ceb70e5cc7fbf4844c3381fd7fb4a886e/pkg/kubelet/volumemanager/cache/desired_state_of_world.go#L167).

Reply:
Will look into it!

Reply:
The calling function is grabbing the volumeSpec from podToMount. The issue is that the desired_state_of_world_populator doesnt have access to the volumePluginMgr which is why the plugin lookup happens in DeletePodFromVolume.
https://github.com/kubernetes/kubernetes/pull/57766/files#diff-94e527d26427ceb4f990f05f76952656R246

Reply:
I'm not sure if I understand what you mean. I mean we can find the `podToMount` object in [here](https://github.com/kubernetes/kubernetes/blob/a1527ed3a375b3f289877202052324554293db5c/pkg/kubelet/volumemanager/cache/desired_state_of_world.go#L270), then we can use `podToMount.spec` instead of the `volumeSpec` parameter of `DeletePodFromVolume `.
The benefit of this approach is we needn't to change the signature of `DesiredStateOfWorld.DeletePodFromVolume`, which will reduce some work.

Reply:
I agree with @mlmhl. The `volumeSpec` is already available in the cache, it doesn't need to be passed in.

Reply:
Maybe I am misunderstanding something. Will take another look.

Reply:
My bad... that's what happens when you drop something before a 2 week Christmas break and come back to it afterward. Will have a fix for this.